Example App Crashes when Suspending after Authentication

## Expected Behavior
App does not crash after suspending after login.

## Actual Behavior
When trying to demo some SSO, the example app crashes after login when you suspend the app.


## Steps to Reproduce the Problem

  1. Login.
  2. Press the home button or the menu button on the device.
  3. App crashes.

## Cause

```
02-15 09:03:57.953 6191-6191/org.aerogear.mobile.example E/AndroidRuntime: FATAL EXCEPTION: main
                                                                           Process: org.aerogear.mobile.example, PID: 6191
                                                                           java.lang.RuntimeException: Parcelable encountered IOException writing serializable object (name = org.aerogear.mobile.auth.user.UserPrincipalImpl)
                                                                               at android.os.Parcel.writeSerializable(Parcel.java:1761)
                                                                               at android.os.Parcel.writeValue(Parcel.java:1709)
                                                                               at android.os.Parcel.writeArrayMapInternal(Parcel.java:777)
                                                                               at android.os.BaseBundle.writeToParcelInner(BaseBundle.java:1506)
                                                                               at android.os.Bundle.writeToParcel(Bundle.java:1181)
                                                                               at android.os.Parcel.writeBundle(Parcel.java:817)
                                                                               at android.support.v4.app.FragmentState.writeToParcel(FragmentState.java:120)
                                                                               at android.os.Parcel.writeTypedArray(Parcel.java:1401)
                                                                               at android.support.v4.app.FragmentManagerState.writeToParcel(FragmentManager.java:617)
                                                                               at android.os.Parcel.writeParcelable(Parcel.java:1730)
                                                                               at android.os.Parcel.writeValue(Parcel.java:1636)
                                                                               at android.os.Parcel.writeArrayMapInternal(Parcel.java:777)
                                                                               at android.os.BaseBundle.writeToParcelInner(BaseBundle.java:1506)
                                                                               at android.os.Bundle.writeToParcel(Bundle.java:1181)
                                                                               at android.app.IActivityManager$Stub$Proxy.activityStopped(IActivityManager.java:4617)
                                                                               at android.app.ActivityThread$StopInfo.run(ActivityThread.java:3998)
                                                                               at android.os.Handler.handleCallback(Handler.java:789)
                                                                               at android.os.Handler.dispatchMessage(Handler.java:98)
                                                                               at android.os.Looper.loop(Looper.java:164)
                                                                               at android.app.ActivityThread.main(ActivityThread.java:6541)
                                                                               at java.lang.reflect.Method.invoke(Native Method)
                                                                               at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:240)
                                                                               at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:767)
                                                                            Caused by: java.io.NotSerializableException: org.aerogear.mobile.auth.user.UserRole
                                                                               at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1233)
                                                                               at java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:347)
                                                                               at java.util.HashSet.writeObject(HashSet.java:287)
                                                                               at java.lang.reflect.Method.invoke(Native Method)
                                                                               at java.io.ObjectStreamClass.invokeWriteObject(ObjectStreamClass.java:977)
                                                                               at java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1545)
                                                                               at java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1481)
                                                                               at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1227)
                                                                               at java.io.ObjectOutputStream.defaultWriteFields(ObjectOutputStream.java:1597)
                                                                               at java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1558)
                                                                               at java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1481)
                                                                               at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1227)
                                                                               at java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:347)
                                                                               at android.os.Parcel.writeSerializable(Parcel.java:1756)
                                                                               at android.os.Parcel.writeValue(Parcel.java:1709) 
                                                                               at android.os.Parcel.writeArrayMapInternal(Parcel.java:777) 
                                                                               at android.os.BaseBundle.writeToParcelInner(BaseBundle.java:1506) 
                                                                               at android.os.Bundle.writeToParcel(Bundle.java:1181) 
                                                                               at android.os.Parcel.writeBundle(Parcel.java:817) 
                                                                               at android.support.v4.app.FragmentState.writeToParcel(FragmentState.java:120) 
                                                                               at android.os.Parcel.writeTypedArray(Parcel.java:1401) 
                                                                               at android.support.v4.app.FragmentManagerState.writeToParcel(FragmentManager.java:617) 
                                                                               at android.os.Parcel.writeParcelable(Parcel.java:1730) 
                                                                               at android.os.Parcel.writeValue(Parcel.java:1636) 
                                                                               at android.os.Parcel.writeArrayMapInternal(Parcel.java:777) 
                                                                               at android.os.BaseBundle.writeToParcelInner(BaseBundle.java:1506) 
                                                                               at android.os.Bundle.writeToParcel(Bundle.java:1181) 
                                                                               at android.app.IActivityManager$Stub$Proxy.activityStopped(IActivityManager.java:4617) 
                                                                               at android.app.ActivityThread$StopInfo.run(ActivityThread.java:3998) 
                                                                               at android.os.Handler.handleCallback(Handler.java:789) 
                                                                               at android.os.Handler.dispatchMessage(Handler.java:98) 
                                                                               at android.os.Looper.loop(Looper.java:164) 
                                                                               at android.app.ActivityThread.main(ActivityThread.java:6541) 
                                                                               at java.lang.reflect.Method.invoke(Native Method) 
                                                                               at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:240) 
                                                                               at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:767) 

```

--- a/auth/src/main/java/org/aerogear/mobile/auth/user/RoleType.java
+++ b/auth/src/main/java/org/aerogear/mobile/auth/user/RoleType.java
@@ -1,8 +1,10 @@
 package org.aerogear.mobile.auth.user;
 
+import java.io.Serializable;
+
 /**
  * Keycloak role types
  */
-public enum RoleType {
+public enum RoleType implements Serializable {
     REALM, CLIENT
 }

--- a/auth/src/main/java/org/aerogear/mobile/auth/user/UserRole.java
+++ b/auth/src/main/java/org/aerogear/mobile/auth/user/UserRole.java
@@ -1,11 +1,12 @@
 package org.aerogear.mobile.auth.user;
 
+import java.io.Serializable;
 import java.util.Objects;
 
 /**
  * Represents a user's keycloak roles information.
  */
-public class UserRole {
+public class UserRole implements Serializable {
 
     /**
      * Role name. Can't be null.

